runner/src/main.rs

```diff
@@ -650,10 +658,39 @@ fn run_example(opt: &RunExamples, example: &Example) -> Step {
     #[allow(clippy::collapsible_if)]
     let run_backend_server_clients: Step = if opt.run_server.unwrap_or(true) {
         let run_server_clients = if opt.build_client.client_variant != NO_CLIENTS {
-            Step::WithBackground {
-                name: "background server".to_string(),
-                background: run_server,
-                foreground: Box::new(run_clients),
+            if example.rerun_server_for_each_client {
+                Step::Multiple {
+                    name: "run separate clients".to_string(),
+                    steps: example
+                        .clients
+                        .iter()
+                        .filter(|(name, _)| match opt.build_client.client_variant.as_str() {
+                            ALL_CLIENTS => true,
+                            client => *name == client,
+                        })
+                        .map(|(name, client)| Step::WithBackground {
+                            name: "background server".to_string(),
+                            background: run_example_server(
+                                &opt.build_server,
```

Comment:
Is this basically a slightly modified version of `run_clients` above? If so, how would future maintainers realise this without a comment or some other indication of what's going on here?